### PR TITLE
development: /s/name/pname&version/

### PR DESCRIPTION
**Pull request by Stunkymonkey**
###### Motivation for this change
#103997
continuation of  #126993

###### Things done

- [ ] Tested using sandboxing ([nix.useSandbox](https://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](https://nixos.org/nix/manual/#sec-conf-file) on non-NixOS linux)
- Built on platform(s)
   - [ ] NixOS
   - [ ] macOS
   - [ ] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [ ] Tested compilation of all pkgs that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review wip"`
- [ ] Tested execution of all binary files (usually in `./result/bin/`)
- [21.11 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md#generating-2111-release-notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

### CODE REVIEW

**Comment by Stunkymonkey** [comment]
is this preferred to be merged into `master`?

**Comment by SuperSandro2000** [comment]
> is this preferred to be merged into `master`?

If we can't separate the mass rebuild we need to target staging.

File: pkgs/development/libraries/imlib/default.nix [hunk 1]
```diff
@@ -2,10 +2,11 @@
...
 }:
 
-stdenv.mkDerivation {
-  name = "imlib-1.9.15";
+stdenv.mkDerivation rec {
+  pname = "imlib";
+  version = "1.9.15";
   src = fetchurl {
-    url = "http://tarballs.nixos.org/imlib-1.9.15.tar.gz";
+    url = "http://tarballs.nixos.org/imlib-${version}.tar.gz";
```
Comment (SuperSandro2000) [inline pkgs/development/libraries/imlib/default.nix:9]:
```suggestion
    url = "https://tarballs.nixos.org/imlib-${version}.tar.gz";
```

File: pkgs/development/libraries/hawknl/default.nix [hunk 2]
```diff
@@ -1,7 +1,8 @@
 {lib, stdenv, fetchurl, unzip}:
 
 stdenv.mkDerivation {
-  name = "hawknl-1.68";
+  pname = "hawknl";
+  version = "1.68";
 
   src = fetchurl {
     url = "http://urchlay.naptime.net/~urchlay/src/HawkNL168src.zip";
```
Comment (SuperSandro2000) [inline pkgs/development/libraries/hawknl/default.nix:8]: The version number can be substituted with replaceString.

File: pkgs/development/libraries/gtkimageview/default.nix [hunk 3]
```diff
@@ -1,10 +1,11 @@
 { fetchurl, lib, stdenv, pkg-config, gtk2 }:
 
 stdenv.mkDerivation rec {
-  name = "gtkimageview-1.6.4";
+  pname = "gtkimageview";
+  version = "1.6.4";
 
   src = fetchurl {
-    url = "http://trac.bjourne.webfactional.com/chrome/common/releases/${name}.tar.gz";
+    url = "http://trac.bjourne.webfactional.com/chrome/common/releases/${pname}-${version}.tar.gz";
```
Comment (SuperSandro2000) [inline pkgs/development/libraries/gtkimageview/default.nix:8]: Does not resolve for me.

File: pkgs/development/libraries/gnet/default.nix [hunk 4]
```diff
@@ -1,7 +1,8 @@
 {lib, stdenv, fetchFromGitHub, pkg-config, autoconf, automake, glib, libtool }:
 
 stdenv.mkDerivation {
-  name = "gnet-2.0.8";
+  pname = "gnet";
+  version = "2.0.8";
   src = fetchFromGitHub {
     owner = "GNOME";
     repo = "gnet";
```
Comment (SuperSandro2000) [inline pkgs/development/libraries/gnet/default.nix:8]: Please substitute the version number.

File: pkgs/development/libraries/cloog-ppl/default.nix [hunk 5]
```diff
@@ -1,10 +1,11 @@
 { fetchurl, lib, stdenv, ppl, autoreconfHook }:
 
 stdenv.mkDerivation rec {
-  name = "cloog-ppl-0.15.11";
+  pname = "cloog-ppl-0.15.11";
+  version = "cloog-ppl-0.15.11";
```
Comment (SuperSandro2000) [inline pkgs/development/libraries/cloog-ppl/default.nix:5]:
```suggestion
  pname = "cloog-ppl";
  version = "0.15.11";
```

File: pkgs/development/libraries/amrnb/default.nix [hunk 6]
```diff
@@ -1,7 +1,8 @@
 {lib, stdenv, fetchurl, unzip}:
 
 stdenv.mkDerivation {
-  name = "amrnb-11.0.0.0";
+  pname = "amrnb";
+  version = "11.0.0.0";
   srcAmr = fetchurl {
     url = "http://www.3gpp.org/ftp/Specs/latest/Rel-11/26_series/26104-b00.zip";
```
Comment (SuperSandro2000) [inline pkgs/development/libraries/amrnb/default.nix:7]:
```suggestion
    url = "https://www.3gpp.org/ftp/Specs/latest/Rel-11/26_series/26104-b00.zip";
```

File: pkgs/development/libraries/agg/default.nix [hunk 7]
```diff
@@ -2,9 +2,10 @@
 , freetype, SDL, libX11 }:
 
 stdenv.mkDerivation rec {
-  name = "agg-2.5";
+  pname = "agg";
+  version = "2.5";
   src = fetchurl {
-    url = "http://www.antigrain.com/${name}.tar.gz";
+    url = "http://www.antigrain.com/${pname}-${version}.tar.gz";
```
Comment (SuperSandro2000) [inline pkgs/development/libraries/agg/default.nix:8]:
```suggestion
    url = "https://www.antigrain.com/${pname}-${version}.tar.gz";
```

**Review by SuperSandro2000 — CHANGES_REQUESTED** [review]

**Comment by SuperSandro2000** [comment]
Feel free to create a todo issue for any of the fetchFromGitHub converts if it gets to much for you.

File: pkgs/development/libraries/liblqr-1/default.nix [hunk 8]
```diff
@@ -1,10 +1,11 @@
 { lib, stdenv, fetchurl, pkg-config, glib }:
 
 stdenv.mkDerivation rec {
-  name = "liblqr-1-0.4.2";
+  pname = "liblqr-1";
+  version = "0.4.2";
 
   src = fetchurl {
-    url = "${meta.homepage}/local--files/en:download-page/${name}.tar.bz2";
+    url = "http://liblqr.wikidot.com/local--files/en:download-page/${pname}-${version}.tar.bz2";
```
Comment (SuperSandro2000) [inline pkgs/development/libraries/liblqr-1/default.nix:8]:
```suggestion
    url = "https://liblqr.wdfiles.com/local--files/en%3Adownload-page/${pname}-${version}.tar.bz2";
```

Comment (Stunkymonkey) [reply pkgs/development/libraries/liblqr-1/default.nix:8]: maybe we should change it to github? https://github.com/carlobaldassi/liblqr

File: pkgs/development/libraries/gtkimageview/default.nix (hunk 3 above)
Comment (Stunkymonkey) [reply pkgs/development/libraries/gtkimageview/default.nix:8]: maybe we could use the link from [arch-linux](https://github.com/archlinux/svntogit-community/blob/packages/gtkimageview/trunk/PKGBUILD#L14)?

**Comment by Stunkymonkey** [comment]
> Feel free to create a todo issue for any of the fetchFromGitHub converts if it gets to much for you.

I think the next PRs will include no `fetchFromGitHub` changes. I will make those in seperate PRs to `master`.

File: pkgs/development/libraries/gtkimageview/default.nix (hunk 3 above)
Comment (SuperSandro2000) [reply pkgs/development/libraries/gtkimageview/default.nix:8]: We can do that if you want.

File: pkgs/development/libraries/liblqr-1/default.nix (hunk 8 above)
Comment (SuperSandro2000) [reply pkgs/development/libraries/liblqr-1/default.nix]: They announced that here http://liblqr.wikidot.com/forum/t-905211/code-hosting-moved-to-github which is linked from http://liblqr.wikidot.com/ so we can do that.

I just pasted the link I got when I visited the download page but you are right. Not sure what happened here.

File: pkgs/development/libraries/science/math/openspecfun/default.nix [hunk 9]
```diff
@@ -1,10 +1,13 @@
-{ lib, stdenv, fetchurl, gfortran }:
+{ lib, stdenv, fetchFromGitHub, gfortran }:
 
-stdenv.mkDerivation {
-  name = "openspecfun-0.5.3";
-  src = fetchurl {
-    url = "https://github.com/JuliaLang/openspecfun/archive/v0.5.3.tar.gz";
-    sha256 = "1rs1bv8jq751fv9vq79890wqf9xlbjc7lvz3ighzyfczbyjcf18m";
+stdenv.mkDerivation rec {
+  name = "openspecfun";
```
Comment (SuperSandro2000) [inline pkgs/development/libraries/science/math/openspecfun/default.nix:4]:
```suggestion
  pname = "openspecfun";
```

**Review by SuperSandro2000 — CHANGES_REQUESTED** [review]

**Review by SuperSandro2000 — APPROVED** [review]

**Comment by Stunkymonkey** [comment]
i know, that this has many rebuilds, but is there anything more to do?

File: pkgs/development/libraries/libxmp/default.nix [hunk 10]
```diff
@@ -16,7 +17,12 @@ stdenv.mkDerivation rec {
   };
 
   src = fetchurl {
+<<<<<<< HEAD
     url = "mirror://sourceforge/xmp/libxmp/${name}.tar.gz";
     sha256 = "sha256-eEfSYhEtFOhEL0TlrG7Z3bylTCUShHILVjyFKzHybnU=";
+=======
+    url = "mirror://sourceforge/xmp/libxmp/${pname}-${version}.tar.gz";
+    sha256 = "1kycz4jsyvmf7ny9227b497wc7y5ligydi6fvvldmkf8hk63ad9m";
+>>>>>>> b4f57a20082 (development: /s/name/pname&version/)
```
Comment (SuperSandro2000) [inline pkgs/development/libraries/libxmp/default.nix:26]: You messed up here.

File: pkgs/development/libraries/boolstuff/default.nix [hunk 11]
```diff
@@ -3,10 +3,11 @@
 let baseurl = "https://perso.b2b2c.ca/~sarrazip/dev"; in
 
 stdenv.mkDerivation rec {
```
Comment (SuperSandro2000) [inline pkgs/development/libraries/boolstuff/default.nix:5]:
```suggestion
stdenv.mkDerivation rec {
```

File: pkgs/development/libraries/boolstuff/default.nix [hunk 12]
```diff
@@ -3,10 +3,11 @@
...
-  name = "boolstuff-0.1.16";
+  pname = "boolstuff";
+  version = "0.1.16";
 
   src = fetchurl {
-    url = "${baseurl}/${name}.tar.gz";
+    url = "${baseurl}/${pname}-${version}.tar.gz";
```
Comment (SuperSandro2000) [inline pkgs/development/libraries/boolstuff/default.nix:10]:
```suggestion
    url = "https://perso.b2b2c.ca/~sarrazip/dev/${pname}-${version}.tar.gz";
```

**Review by SuperSandro2000 — CHANGES_REQUESTED** [review]

**Comment by siraben** [comment]
Building the minimal ISO

```nix
$ nix-build ./nixos/release-combined.nix -A nixos.iso_minimal.x86_64-linux
```

File: pkgs/development/libraries/portaudio/default.nix [hunk 13]
```diff
@@ -1,11 +1,12 @@
...
 
-stdenv.mkDerivation {
-  name = "portaudio-190600-20161030";
+stdenv.mkDerivation rec {
+  pname = "portaudio";
+  version = "190600-20161030";
 
   src = fetchurl {
-    url = "http://www.portaudio.com/archives/pa_stable_v190600_20161030.tgz";
+    url = "http://www.portaudio.com/archives/pa_stable_v${version}.tgz";
```
Comment (jtojnar) [inline pkgs/development/libraries/portaudio/default.nix:9]: Replace dash with underscore in the URL.

**Review by siraben — APPROVED** [review]
Minimal ISO builds.

File: pkgs/development/libraries/portaudio/default.nix (hunk 13 above)
Comment (Stunkymonkey) [reply pkgs/development/libraries/portaudio/default.nix]: oh yes. very good that you found this little mistake. fixed it

**Pull request merged**